pkg/scheduler/framework/plugins/noderesources/resource_allocation.go

```diff
@@ -51,17 +58,18 @@ func (r *resourceAllocationScorer) score(
 	if node == nil {
 		return 0, framework.NewStatus(framework.Error, "node not found")
 	}
-	if r.resourceToWeightMap == nil {
+	if r.weightList == nil {
 		return 0, framework.NewStatus(framework.Error, "resources not found")
 	}
 
-	requested := make(resourceToValueMap)
-	allocatable := make(resourceToValueMap)
-	for resource := range r.resourceToWeightMap {
-		alloc, req := r.calculateResourceAllocatableRequest(nodeInfo, pod, resource)
+	requested := make(valueList, 0, len(r.weightList))
+	allocatable := make(valueList, 0, len(r.weightList))
+	for i := range r.weightList {
+		alloc, req := r.calculateResourceAllocatableRequest(nodeInfo, pod, r.resourceNameList[i])
```

Comment:
for a follow up PR:

`calculatePodResourceRequest` could be done in PreScore, for an extra performance boost.

Reply:
I'd like to do.